packages/cardano-services/src/StakePoolSearch/DbSyncStakePoolSearchProvider/queries.ts

```diff
@@ -91,15 +328,16 @@ FROM pool_relay
 WHERE update_id = ANY($1)
 `;
 
-// TODO: probably this should be filtered by registration tx_id = last_update.registration_tx_id
 export const findPoolsOwners = `
 SELECT 
-  DISTINCT stake.view AS address,
+  stake.view AS address,
   owner.pool_hash_id AS hash_id
 FROM pool_owner owner
+JOIN pool_update 
+    ON pool_update.registered_tx_id = owner.registered_tx_id
```

Comment:
@rhyslbw @mkazlauskas Can you please double check this.

> Pools can be registered, then retired and then registered again. What happens if the owners of the first pool registration are different from the last registration? Should we filter them? Thats why I put this comment. We currently have the tx where the owner was registered and this should match with the pool's registration.

Reply:
Hm I'm not sure. May be best to ask db-sync devs how this scenario is handled so we know how to query it.

Reply:
> What happens if the owners of the first pool registration are different from the last registration?

@james-iohk is this a legal state change?

Reply:
> is this a legal state change?

Having thought about this more, we don't need to be concerned with this. As @mkazlauskas suggested, the last pool update owner is all we care about.

Reply:
Yeah, that is what this change does. So I think we are good :)